pkg/cloud/azure/provider.go

```diff
@@ -290,9 +290,10 @@ func getRetailPrice(region string, skuName string, currencyCode string, spot boo
 	retailPrice := ""
 	for _, item := range pricingPayload.Items {
 		if item.Type == "Consumption" && !strings.Contains(item.ProductName, "Windows") {
-			// if spot is true SkuName should contain "spot, if it is false it should not
 			if spot == strings.Contains(strings.ToLower(item.SkuName), " spot") {
 				retailPrice = fmt.Sprintf("%f", item.RetailPrice)
+			} else {
+				retailPrice = fmt.Sprintf("%f", item.RetailPrice)
```

Comment:
can you re do this so that in the case of a spot instance it looks for a spot sku first and if there is not one it returns ondemand and for ondemand it should never be able to return a spot sku

Reply:
Makes total sense, don't know why this change was in the PR